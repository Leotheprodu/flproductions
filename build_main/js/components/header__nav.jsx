import { useEffect, useState } from 'react';
import { NavLink, Link } from 'react-router-dom';



export function NavMenu() {

  const [data, setData] = useState(null);

  useEffect(() => {
    fetch("/api")
      .then((res) => res.json())
      .then((data) => setData(data.message));
  }, []);

  return(

          <div className='header__nav'>

            <Link to="/" className='header__web-tittle'>{!data ? "FLP" : data}</Link>

            <nav className='header__links'>
              <NavLink to='/' end className='header__links__link'>Inicio</NavLink>
              <NavLink to='/nosotros' className='header__links__link'>Nosotros</NavLink>
              <NavLink to='/contacto' className='header__links__link'>Contacto</NavLink>
              {/* <NavLink to='/blog' className='header__links__link'>Blog</NavLink>
              <NavLink to='/servicios' className='header__links__link'>Servicios</NavLink> */}

            </nav>
            

          </div>

  

  )

}