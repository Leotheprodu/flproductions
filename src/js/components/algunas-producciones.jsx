import { IconBrandInstagram } from "@tabler/icons";
import { useState } from "react";
import { algunasProducciones } from "./database/database";
import { YoutubeEmbed } from "./Helpers/youtubeEmbed";

export function AlgunasProducciones() {
    const [youtubeLink, setyoutubeLink] = useState('');
    const [instagramLink, setinstagramLink] = useState('');
    const [artistaItem, setartistaItem] = useState('');
    const [nombreItem, setnombreItem] = useState('');
    const [descripcionItem, setdescripcionItem] = useState('');
    const [onClick, setonClick] = useState(false);

    const prueba = (youtube, instagram, artista, descripcion, nombre) => {
        setyoutubeLink(youtube);
        setonClick(true);
        setinstagramLink(instagram);
        setartistaItem(artista);
        setdescripcionItem(descripcion);
        setnombreItem(nombre);
    }


    return(
        <div className="algunas-producciones">

            <h2>Algunas producciones hechas por nosotros</h2>
            

            <div className="algunas-producciones__botones">

                {
                    algunasProducciones.map(({ nombre, artista, youtube, instagram, descripcion }) => (
                        <div className="algunas-producciones__boton" key={ youtube }>
                            <p onClick={ () => prueba( youtube, instagram, artista, descripcion, nombre ) }>{ nombre }</p>
                        </div>
                    ))
                }
            </div>
            {
                onClick &&
                <div className="algunas-producciones__detalles contenedor">
                <div className="algunas-producciones__detalles__texto">

                    <h3>Nombre: <span>{nombreItem}</span>, interpretado por: <span>{artistaItem}</span></h3>

                    <p>{descripcionItem}</p>
                </div>
                
                <div className="algunas-producciones__detalles__instagram">
                    
                    <a target='_blank' href={instagramLink}>
                        <IconBrandInstagram size={50}/>
                        <p>{artistaItem}</p>
                    </a>
                    <a target='_blank' href='https://www.instagram.com/leotheprodu/'>
                        <IconBrandInstagram size={50}/>
                        <p>LeotheProdu</p>
                    </a>
                </div>
                </div>
            }

            <div  className="contenedor algunas-producciones__videoDiv">
                <YoutubeEmbed embedId={youtubeLink}/>
            </div>

        </div>

    )


}