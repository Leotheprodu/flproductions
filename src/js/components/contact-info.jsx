



export function ContactInfo({icon, titulo, parrafo}) {

    return (
        <div className="contact-info">
        
            <div className="contact-info__icon">
                {icon}
            </div>
            <div>
            <h3 className="contact-info__titulo">{titulo}</h3>
                <p className="contact-info__parrafo">{parrafo}</p>
            </div>
                
        </div>

    )

}