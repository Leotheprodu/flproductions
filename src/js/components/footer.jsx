import { SocialIcons } from "./social-icons";



export function Footer () {

    const copyright = new Date();

    const actualYear = copyright.getFullYear();
    
    return(

        <div className="footer">

            <div>
                <SocialIcons size={24} claseCSS='header__socialIcons__icons' />
            </div>

            <p>Â© 2013 - {actualYear} FLProductions - creado por LeotheProdu</p>

        </div>

    )
}