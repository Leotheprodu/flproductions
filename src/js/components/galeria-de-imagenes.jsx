import { useState } from "react";
import PhotoAlbum from "react-photo-album"
import Lightbox from "yet-another-react-lightbox";
// import optional lightbox plugins
import Fullscreen from "yet-another-react-lightbox/plugins/fullscreen";
import Slideshow from "yet-another-react-lightbox/plugins/slideshow";
import Zoom from "yet-another-react-lightbox/plugins/zoom";
import Thumbnails from "yet-another-react-lightbox/plugins/thumbnails";
import "yet-another-react-lightbox/plugins/thumbnails.css";
import "yet-another-react-lightbox/styles.css";
import { fotosDelEstudio } from "./database/database";



export function GaleriaDeImagenes() {

    const [open, setOpen] = useState(false);
    const [NuevoArray, setNuevoArray] = useState(fotosDelEstudio);

    
    function imagenClick(e) {
        setOpen(true);
        let primerImagen = e.target.src;
        const filtrados = fotosDelEstudio.filter( item => item.src !== primerImagen);
        setNuevoArray([
            {
            src: primerImagen,

            },...filtrados
        ]);

    };
return (
<div className="galeria-de-imagenes">
    <h3>Fotos del estudio</h3>
    <PhotoAlbum padding={10} onClick={imagenClick} layout="rows" photos={fotosDelEstudio} />
    <Lightbox
        open={open}
        close={() => setOpen(false)}
        slides={NuevoArray}
        plugins={[Thumbnails, Fullscreen, Slideshow, Zoom]}
    />
</div>

)


}