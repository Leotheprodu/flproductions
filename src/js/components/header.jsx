import PropTypes from 'prop-types'

export function Header ({imgName, pagina}) {

  return (

    <div className={"header"+" "+pagina+"-contenedor"}>

      <div>
        
        <picture>

          <img className={pagina+"-img"} style={{ backgroundImage:`url(${"build_main/img/"+imgName})` }} loading="lazy" />
        </picture>

      </div>


    </div>

  )

}
Header.propTypes = {
    
  pagina: PropTypes.string.isRequired,
  imgName: PropTypes.string.isRequired,
  
}
Header.defaultProps = {

  imgName: "header-main",
  pagina: "default",

}