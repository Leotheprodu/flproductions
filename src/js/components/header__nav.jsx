import propTypes from 'prop-types'
import { BlogRender } from '../pages/blog';
import { ContactoRender } from '../pages/contacto';
import { InicioRender } from '../pages/inicio';
import { NosotrosRender } from '../pages/nosotros';
import { ServiciosRender } from '../pages/servicios';

let claseInteligente = 'nav-normal'


export function NavMenu({pagina}) {

  function inicio() {
    if(pagina ==='inicio') {

      return "nav_menu-active"
    }
  }
  function nosotros() {
    if(pagina ==='nosotros') {

      return "nav_menu-active"
    }
  }
  function blog() {
    if(pagina ==='blog') {

      return "nav_menu-active"
    }
  }
  function contacto() {
    if(pagina ==='contacto') {

      return "nav_menu-active"
    }
  }
  function servicios() {
    if(pagina ==='servicios') {

      return "nav_menu-active"
    }
  }

  return(

    <div className='header__nav'>

      <a href="#" onClick={InicioRender} className='header__web-tittle'>FLProductions</a>

      <nav>

        <a onClick={InicioRender} className={claseInteligente+" "+ inicio()} href="#">Inicio</a>
        <a onClick={NosotrosRender} className={claseInteligente+" "+ nosotros()} href="#">Nosotros</a>
        <a onClick={BlogRender} className={claseInteligente+" "+ blog()} href="#">Blog</a>
        <a onClick={ContactoRender} className={claseInteligente+" "+ contacto()} href="#">Contacto</a>
        <a onClick={ServiciosRender} className={claseInteligente+" "+ servicios()} href="#">Servicios</a>
        

      </nav>
      

    </div>


  )


}


NavMenu.propTypes = {
    
  pagina: propTypes.string.isRequired,
  
}