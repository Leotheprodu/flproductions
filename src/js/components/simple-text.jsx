import propTypes from 'prop-types';
import {useRef, useEffect} from 'react';





export function SimpleText({titulo,texto,fxElement,tipo}) {

    const ref = useRef(null);
    
    useEffect(() => {
      const el1 = ref.current;
      
      function mostrarScroll(){
        let scrollTop = document.documentElement.scrollTop
        let alturaAnimado = el1.offsetTop;
        if (alturaAnimado - 600 < scrollTop) {
          el1.style.opacity = 1;
          if(tipo !== 1 && fxElement === 'fxElement') {
            el1.classList.add('fxMostrarArriba')
          }
          
        }
        
      }
    
      
      window.addEventListener('scroll',mostrarScroll);
      
    },[]);
    
  if(tipo===1) {
    return(

      <div className={'simple-text_info'}>
        <h1>{titulo}</h1>

        <p>{texto}</p>
      </div>
    );

  }else if(tipo===2) {
    
    return(

    
      <div /* onScroll={mostrarScroll} */ ref={ref} className={'simple-text_info'+' '+fxElement}>
        <h3>{titulo}</h3>

        <p>{texto}</p>
      </div>
    );
  };


};


SimpleText.propTypes = {
    
  titulo: propTypes.string.isRequired,
  tipo: propTypes.number.isRequired, //1: contiene un h1 y 2: contiene un h3
  texto: propTypes.string.isRequired,
  fxElement: propTypes.string,
  
}
SimpleText.defaultProps = {

  tipo: 2,

}