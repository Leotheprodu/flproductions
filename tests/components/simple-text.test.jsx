
import { fireEvent, render, screen } from "@testing-library/react";
import { SimpleText } from "../../src/js/components/simple-text";

describe('SimpleText pruebas locas', () => {

    const clase='fxElement'
    const titulo='Acerca de Nosotros' 
    const texto='Somos algo más que sólo un estudio de grabación en Costa Rica, no nos limitamos a hacer únicamente nuestro trabajo, queremos que en cada proyecto, cada canción lleve nuestro ADN, damos siempre lo mejor de nosotros y definitivamente hacer esto, es nuestra misión de vida.' 

test('crear snapshot', () => {


    const { container } = render( <SimpleText titulo={titulo} fxElement={clase} texto={texto} /> );
    expect( container ).toMatchSnapshot();

});
    
});