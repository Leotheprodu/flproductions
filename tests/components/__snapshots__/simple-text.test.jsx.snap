// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleText pruebas locas crear snapshot 1`] = `
<div>
  <div
    class="simple-text_info fxElement"
  >
    <h3>
      Acerca de Nosotros
    </h3>
    <p>
      Somos algo más que sólo un estudio de grabación en Costa Rica, no nos limitamos a hacer únicamente nuestro trabajo, queremos que en cada proyecto, cada canción lleve nuestro ADN, damos siempre lo mejor de nosotros y definitivamente hacer esto, es nuestra misión de vida.
    </p>
  </div>
</div>
`;
